dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModelManager.java

```diff
@@ -0,0 +1,74 @@
+package com.dat3m.dartagnan.verification.model;
+
+import com.dat3m.dartagnan.encoding.EncodingContext;
+import com.dat3m.dartagnan.program.memory.MemoryObject;
+import com.dat3m.dartagnan.verification.model.event.EventModelManager;
+import com.dat3m.dartagnan.verification.model.relation.RelationModelManager;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.Formula;
+import org.sosy_lab.java_smt.api.Model;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+
+import java.math.BigInteger;
+import java.util.List;
+
+
+public class ExecutionModelManager {
+    private final EventModelManager eMManager;
+    private final RelationModelManager rMManager;
+
+    private EncodingContext context;
+    private Model model;
+    private ExecutionModelNext executionModel;
+
+    public ExecutionModelManager(){
+        eMManager = new EventModelManager(this);
+        rMManager = new RelationModelManager(this);
+    }
+
+    public ExecutionModelNext buildExecutionModel(EncodingContext context, Model model) {
+        if (this.context != null && this.model != null) {
+            if (this.context == context && this.model == model) {
+                return executionModel;
+            }
+        }
+
+        this.context = context;
+        this.model = model;
+        executionModel = new ExecutionModelNext(this);
+
+        eMManager.buildEventModels(executionModel, context);
+        extractMemoryLayout();
+        rMManager.buildRelationModels(executionModel, context);
+
+        return executionModel;
+    }
+
+    public EncodingContext getContext() {
+        return context;
+    }
+
+    public boolean isTrue(BooleanFormula formula) {
+        return Boolean.TRUE.equals(model.evaluate(formula));
+    }
+
+    public Object evaluateByModel(Formula formula) {
+        return model.evaluate(formula);
+    }
+
+    public void extractRelations(List<String> relationNames) {
+        rMManager.extractRelations(relationNames);
+    }
```

Comment:
Do these methods need to be public? What's the purpose of the last one (it returns nothing)?

Reply:
It serves as the API used by `ExecutionGraphVisualizer` to extract `Relations` to be shown on demand, and then add the constructed `RelationModels` to `ExecutionModelNext` which can be queried and visualized.

Reply:
The first two methods are called by `EventModelManager`, `RelationModelManager` and `ExecutionModelManager` itself, that's why I let `Model` be stored by and only by this class.